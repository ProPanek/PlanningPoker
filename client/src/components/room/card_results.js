import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";

import {Card} from "@material-ui/core";


const Wrapper = styled(Card)`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
    margin-top:10px;

  width: 90%;
  flex-wrap: wrap;
`;

const CardsWrapper = styled.div`
  &&{
  margin: 0 auto;
  width: 90%;
  margin-bottom:10px;
  display:flex;
  }
`;

const StyledCard = styled(Card)`
  &&{
    width:57px;
    height:89px;
    font-size: -webkit-xxx-large;
    line-height: 89px;
    margin: 5px;
    margin: 0 auto;
    background-color:${props => props.color || "#303F9F"};
    color:white;
  }
`;


class CardResults extends Component {

  render() {

    const {store: {room: {waiting, cardResults}}} = this.props
    return (
      <React.Fragment>
        <Wrapper>
          <CardsWrapper>
            {waiting.length > 0 && waiting.map((result, index) => (
              <Card key={index} style={{width: "100px", minHeight: "20px"}}>
                <Typography>
                </Typography>
                <StyledCard color={"#303F9F"}>
                </StyledCard>
              </Card>
            ))}

          </CardsWrapper>
          <CardsWrapper>
            {cardResults.length > 0 && cardResults.map((result, index) => (
              <Card key={index} style={{width: "100px", minHeight: "20px"}}>
                <Typography>
                  {result.userName}
                </Typography>
                <StyledCard color={result.color}>
                  {result.cardValue}
                </StyledCard>
              </Card>
            )) || <Typography style={{margin: "0 auto"}}> Waiting for response from all users </Typography>}
          </CardsWrapper>
        </Wrapper>
      </React.Fragment>
    );
  }
}

export default inject("store")(withRouter(observer(CardResults)));

