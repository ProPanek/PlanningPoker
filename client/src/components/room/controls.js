import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Card";
import styled from "styled-components";
import {Button, Card} from "@material-ui/core";
import {decorate, observable} from "mobx";

const StyledButtonCard = styled(Button)`
  &&{
  width: 10%;
  height: 5vh;
  margin:5px;
  flex:1;
  }
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 83.333333%;
  height: 10vh;
  flex-wrap: wrap;
`;

const Wrapper = styled(Card)`
  &&{
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: 90%;
  flex-wrap: wrap;
  margin-top:10px;
  margin-bottom: 10px;
  padding: 5px;
  justify-content: center;
  }
`;
const cards = [0.5, 1, 2, 3, 5, 8, 13, 20, 40, 100]

class Controls extends Component {


  handleReset = () => {
    this.props.store.resetCards()
    this.props.store.cardResults = []
  }

  handleCard = () => {
    this.props.store.sendCard(this.selectedCard)
    this.props.store.blockCard = true
  }

  chooseCard = (card) => {
    this.selectedCard = card
  }

  render() {
    return (
      <React.Fragment>
        <Wrapper> We're waiting for : {this.props.store.waiting} users </Wrapper>
        <Button disabled={this.props.store.blockCard} variant="contained" color="secondary"
                onClick={this.handleCard}>Send Card</Button>
        {this.props.store.admin && (
          <Button onClick={this.handleReset}>Next issue</Button>
        )}
        <ButtonWrapper>
          {cards.map((card, index) => {
            return (
              <StyledButtonCard variant="contained" color="primary" key={index} onClick={() => {
                this.chooseCard(card)
              }}>{card}</StyledButtonCard>
            )
          })}
        </ButtonWrapper>
      </React.Fragment>
    )
  }
}

decorate(Controls, {
  selectedCard: observable,
});


export default inject("store")(withRouter(observer(Controls)));

