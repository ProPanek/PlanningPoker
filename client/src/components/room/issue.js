import React, {Component} from 'react'
import TextField from '@material-ui/core/TextField'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";

import styled from "styled-components";
import ControlledEditor from '../../components/editor/ControlledEditor'

const StyledTextField = styled(TextField)`
  &&{
  width:90%;
  }
`

const StyledCard = styled(Card)`
  &&{
  margin: 0 auto;
  width:95%;
  }
`

const StyledTitleCard = styled(Card)`
  &&{
  margin: 0 auto;
    margin-top:20px;

  width:90%;
  margin-bottom: 15px;
  }
`

class Issue extends Component {


  handleChange = (e) =>{
    if (e.target.id === "title"){
      this.props.store.title = e.target.value
      this.props.store.broadcastTitle()
    }
    if (e.target.id === "description"){
      // this.props.store.description = e.target.value
      // this.props.store.broadcastDescription()
    }
  }

  render() {

    return (
      <StyledCard>
        <StyledTitleCard>
          <Typography variant="subtitle2">
            Title
          </Typography>
          <StyledTextField
            id="title"
            value={this.props.store.title}
            onChange={this.handleChange}
            margin="normal"
          />
        </StyledTitleCard>
        <div>
          <Typography variant="subtitle2">
            Description
          </Typography>
          <ControlledEditor />

        </div>
      </StyledCard>
    )
  }
}


export default inject("store")(withRouter(observer(Issue)));

