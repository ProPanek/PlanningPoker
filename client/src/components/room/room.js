import React, {Component} from "react";
import Card from "@material-ui/core/Card";
import Grid from "@material-ui/core/Grid";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import JoinDialog from '../join_room/join_dialog'
import Issue from './issue'
import CardResults from './card_results'
import Controls from './controls'
import UserList from './user_list'

const StyledGrid = styled(Grid)`
  &&{
  height: calc(100vh - 48px);
  }
`;

const StyledCard = styled(Card)`
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  }
`;

class Room extends Component {
  componentDidMount() {
    this.props.store.fetchUsers()
    window.onpopstate = this.onBackButtonEvent
    setInterval(() => {
      if (this.props.store.kicked) {
        this.props.history.push(`/`)
        this.props.store.kicked = false
      }
      if (!this.props.store.connected) {
        this.props.store.openJoinDialog = true
      }
    }, 250)
  }

  onBackButtonEvent = (e) => {
    e.preventDefault()
    this.props.store.kickUser(this.props.store.userId)
    this.props.store.rooms = []
    this.props.store.users = []
    this.props.store.admin = false
    this.props.store.connected = false
    this.props.store.userName = ""
    this.props.store.userId = ""
    this.props.store.roomName = ""
    this.props.store.roomId = ""
    setTimeout(() => {
      this.props.store.notificationMessage = "You have leaved the Room"
      this.props.store.notificationVariant = "warning"
      this.props.history.push(`/`)
    }, 100)
  }

  render() {
    return (
      <React.Fragment>
        <JoinDialog/>
        <StyledGrid item xs={10}>
          <StyledCard>
            <Issue />
            <CardResults />
            <Controls />
          </StyledCard>
        </StyledGrid>
        <StyledGrid item xs={2}>
          <StyledCard>
            <UserList />
          </StyledCard>
        </StyledGrid>
      </React.Fragment>
    );
  }
}



export default inject("store")(withRouter(observer(Room)));
