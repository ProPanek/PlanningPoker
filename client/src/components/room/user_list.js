import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";
import Button from "@material-ui/core/Button"
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";


const StyledSelect = styled(Select)`
  width:100%;
`

const DefaultSelect = styled.select`
  width:100%;
`

const RoomName = styled.div`
  && {
  margin: 0 auto;
  }
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-wrap: wrap;
`;

const UserDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;
const JiraDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;

class UserList extends Component {
  state = {
    userId: "",
    selectBoardId: "",
    board: "",
    issue: ""
  }

  handleSelect = e => {
    this.setState({userId: e.target.value});
  };

  handleKick = () => {
    if (this.state.userId !== "") {
      this.props.store.kickUser(this.state.userId)
    }
  }
  handleAdmin = () => {
    if (this.state.userId !== "") {
      this.props.store.changeAdmin(this.state.userId)
    }
  }

  selectBoard = (e) => {
    this.props.store.jira.boardId = e.target.value
    this.props.store.selectBoard(e.target.value)
  }

  selectIssue = (e) => {
    const fields = e.target.value.split("_")
    this.props.store.jira.title = fields[0]
    this.props.store.jira.description = fields[1]
    this.props.store.jira.issueId = fields[2]
    this.props.store.broadcastTitle()
    this.props.store.broadcastDescription()
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value});
  };

  render() {
    return (
      <React.Fragment>
        <UserDiv>
          <RoomName variant="body1" color="inherit">
            {this.props.store.user.userName !== "" && <div> User Name: {this.props.store.user.userName}</div>}
            {this.props.store.room.roomName !== "" && <div> Room Name: {this.props.store.room.roomName}</div>}
          </RoomName>
          <Typography>users : {this.props.store.user.users.length}</Typography>
          <DefaultSelect size={this.props.store.user.users.length} onChange={this.handleSelect}>
            {this.props.store.user.users.length > 0 &&
            this.props.store.user.users.map((data, index) => {
              return (
                <option key={index} value={data.userId}>
                  {console.log(data)}
                  {data.userName}
                </option>
              );
            })}
          </DefaultSelect>
          {this.props.store.user.admin && (
            <Wrapper>
              <Button onClick={this.handleKick}>Kick User</Button>
              <Button onClick={this.handleAdmin}>Give admin </Button>
            </Wrapper>
          )}
        </UserDiv>
        <JiraDiv>
          <Typography>Jira Task Picker</Typography>
          {this.props.store.jira.jiraBoards.values.length > 0 &&
          <StyledSelect
            inputProps={{
              name: 'board',
              id: 'board'
            }}
            value={this.state.board} onChange={(e) => {
            this.selectBoard(e);
            this.handleChange(e);
          }}>

            {this.props.store.jira.jiraBoards.values.map((data, index) => {
              return (
                <MenuItem key={index} value={data.id}>
                  {data.name}
                </MenuItem>
              );
            })}
          </StyledSelect>}
          {this.props.store.jira.activeBoard.issues.length > 0 &&
          <StyledSelect
            inputProps={{
              name: 'issue',
              id: 'issue'
            }}
            value={this.state.issue}
            onChange={(e) => {
              this.selectIssue(e);
              this.handleChange(e);
            }}>
            {this.props.store.jira.activeBoard.issues.map((data, index) => {
              return (
                <MenuItem key={index} value={`${data.fields.summary}_${data.fields.description}_${data.id}`}>
                  {data.key} - {data.fields.summary}
                  {console.log(data)}
                </MenuItem>
              );
            })}
          </StyledSelect>}
        </JiraDiv>
      </React.Fragment>
    )
  }
}


export default inject("store")(withRouter(observer(UserList)));

