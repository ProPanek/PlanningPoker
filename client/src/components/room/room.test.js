import React from "react"
import {Room} from "./room"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const noop = () => {}

describe("Room", () => {
  it("renders without errors", () => {
    const store = {
      room: { cardsAreTheSame : false, cardResults : [], cardHistory: []},
      user: {admin: true, users: []},
      jira: {activeBoardFetching: false, activeBoard: { issues: []}},
      fetchUsers: noop
    }

    const component = shallow(<Room store={store} />)
    expect(component).toMatchSnapshot()
  })
})
