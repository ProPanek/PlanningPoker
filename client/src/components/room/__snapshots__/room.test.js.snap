// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room renders without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Room
    store={
      Object {
        "fetchUsers": [Function],
        "jira": Object {
          "activeBoard": Object {
            "issues": Array [],
          },
          "activeBoardFetching": false,
        },
        "room": Object {
          "cardHistory": Array [],
          "cardResults": Array [],
          "cardsAreTheSame": false,
        },
        "user": Object {
          "admin": true,
          "users": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <inject-withRouter(JoinDialog)-with-store />,
        <ForwardRef
          item={true}
          xs={10}
        >
          <ForwardRef>
            <inject-withRouter(Issue)-with-store />
            <inject-withRouter(CardResults)-with-store />
            <inject-withRouter(Controls)-with-store />
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef
          item={true}
          xs={2}
        >
          <ForwardRef>
            <inject-withRouter(UserList)-with-store />
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <inject-withRouter(Issue)-with-store />
            <inject-withRouter(CardResults)-with-store />
            <inject-withRouter(Controls)-with-store />
          </ForwardRef>,
          "item": true,
          "xs": 10,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <inject-withRouter(Issue)-with-store />,
              <inject-withRouter(CardResults)-with-store />,
              <inject-withRouter(Controls)-with-store />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Naked": [Function],
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-kAzzGY",
              "isStatic": true,
              "rules": Array [
                "
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  }
",
              ],
            },
            "displayName": "Styled(WithStyles(Card))",
            "options": Object {
              "name": "MuiCard",
            },
            "render": [Function],
            "styledComponentId": "sc-kAzzGY",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Naked": [Function],
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-cSHVUG",
            "isStatic": true,
            "rules": Array [
              "
  &&{
  height: calc(100vh - 48px);
  }
",
            ],
          },
          "displayName": "Styled(WithStyles(Grid))",
          "options": Object {
            "name": "MuiGrid",
          },
          "render": [Function],
          "styledComponentId": "sc-cSHVUG",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <inject-withRouter(UserList)-with-store />
          </ForwardRef>,
          "item": true,
          "xs": 2,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <inject-withRouter(UserList)-with-store />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Naked": [Function],
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-kAzzGY",
              "isStatic": true,
              "rules": Array [
                "
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  }
",
              ],
            },
            "displayName": "Styled(WithStyles(Card))",
            "options": Object {
              "name": "MuiCard",
            },
            "render": [Function],
            "styledComponentId": "sc-kAzzGY",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Naked": [Function],
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-cSHVUG",
            "isStatic": true,
            "rules": Array [
              "
  &&{
  height: calc(100vh - 48px);
  }
",
            ],
          },
          "displayName": "Styled(WithStyles(Grid))",
          "options": Object {
            "name": "MuiGrid",
          },
          "render": [Function],
          "styledComponentId": "sc-cSHVUG",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <inject-withRouter(JoinDialog)-with-store />,
          <ForwardRef
            item={true}
            xs={10}
          >
            <ForwardRef>
              <inject-withRouter(Issue)-with-store />
              <inject-withRouter(CardResults)-with-store />
              <inject-withRouter(Controls)-with-store />
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef
            item={true}
            xs={2}
          >
            <ForwardRef>
              <inject-withRouter(UserList)-with-store />
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              <inject-withRouter(Issue)-with-store />
              <inject-withRouter(CardResults)-with-store />
              <inject-withRouter(Controls)-with-store />
            </ForwardRef>,
            "item": true,
            "xs": 10,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <inject-withRouter(Issue)-with-store />,
                <inject-withRouter(CardResults)-with-store />,
                <inject-withRouter(Controls)-with-store />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-kAzzGY",
                "isStatic": true,
                "rules": Array [
                  "
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  }
",
                ],
              },
              "displayName": "Styled(WithStyles(Card))",
              "options": Object {
                "name": "MuiCard",
              },
              "render": [Function],
              "styledComponentId": "sc-kAzzGY",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Naked": [Function],
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-cSHVUG",
              "isStatic": true,
              "rules": Array [
                "
  &&{
  height: calc(100vh - 48px);
  }
",
              ],
            },
            "displayName": "Styled(WithStyles(Grid))",
            "options": Object {
              "name": "MuiGrid",
            },
            "render": [Function],
            "styledComponentId": "sc-cSHVUG",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              <inject-withRouter(UserList)-with-store />
            </ForwardRef>,
            "item": true,
            "xs": 2,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <inject-withRouter(UserList)-with-store />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-kAzzGY",
                "isStatic": true,
                "rules": Array [
                  "
  &&{
  height: calc(100vh - 48px);
  overflow-y: auto;
  }
",
                ],
              },
              "displayName": "Styled(WithStyles(Card))",
              "options": Object {
                "name": "MuiCard",
              },
              "render": [Function],
              "styledComponentId": "sc-kAzzGY",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Naked": [Function],
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-cSHVUG",
              "isStatic": true,
              "rules": Array [
                "
  &&{
  height: calc(100vh - 48px);
  }
",
              ],
            },
            "displayName": "Styled(WithStyles(Grid))",
            "options": Object {
              "name": "MuiGrid",
            },
            "render": [Function],
            "styledComponentId": "sc-cSHVUG",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
